src/Microsoft.PowerShell.Commands.Management/commands/management/TestConnectionCommand.cs

```diff
@@ -0,0 +1,223 @@
+            }
+            set
+            {
+                this.delay = value;
+            }
+        }
+
+        /// <summary>Return only boolean value representing success/failure</summary>
+        [Parameter(ParameterSetName = "Default",
+            Mandatory =false,
+            HelpMessage ="Return only boolean value representing success/failure")]
+        public SwitchParameter Quiet
+        {
+            get
+            {
+                return (SwitchParameter)this.quiet;
+            }
+            set
+            {
+                this.quiet = (bool)value;
+            }
+        }
+#endregion
+#region Meat of the class
+        /// <summary> Create the object </summary>
+        protected override void ProcessRecord()
+        {
+            bool quietReply = false;
+            try
+            {
+                IPHostEntry ipHostEntry = Dns.GetHostEntryAsync(this.destination).Result;
+            
+            //If we got this far, somethings going good
+                try
+			    {
+				    for (int i = 0; i < this.count; i++)
+				    {
+					    Ping sender = new Ping();
+					    PingReply reply = sender.SendPingAsync(this.destination, (timeout * 1000), new byte[] { 0xff }, new PingOptions(this.timetolive, true)).Result;
+                    
+					    //If things are good then set our bool to true, otherwise leave it false or whatever it currently is
+					    if (reply.Status == IPStatus.Success)
+					    {
+						    quietReply = true;
+					    }
+
+					    //We want to be chatty
+					    if (!this.quiet)
+					    {
+						    WriteObject(new PingResponse(reply.Status.ToString(), reply.Address.ToString(), reply.RoundtripTime));
+					    }
+				    }
+				    //If we are passing -quiet we only want the one final pass/fail, so here it is
+				    if (this.quiet)
+				    {
+					    WriteObject(quietReply);
+				    }
+			    }
+			    catch (PingException pex)
+			    {
+				    WriteError(new ErrorRecord(pex, pex.InnerException.ToString(), ErrorCategory.NotSpecified, pex.Source));
+			    }
+			    catch (NullReferenceException nrefex)
+			    {
+				    WriteError(new ErrorRecord(nrefex, nrefex.Message, ErrorCategory.ObjectNotFound, nrefex.Source));
+			    }
+			    catch(Exception ex)
+			    {
+                    WriteError(new ErrorRecord(ex, ex.InnerException.ToString(), ErrorCategory.NotSpecified,ex.Source));
+			    }
+            }
+            catch(Exception ex)
+            {
+                WriteError(new ErrorRecord(ex, ex.InnerException.ToString(), ErrorCategory.NotSpecified,ex.Source));
+            }
+            Thread.Sleep(new TimeSpan(0, 0, this.delay));
+        }
+#endregion
+    }
+
+     /// <summary> Object for Ping Response</summary>
+    public class PingResponse
+    {
+        /// <summary>Response Status</summary>
+        public string Status { get; set; }
+        /// <summary>Remote system being checked</summary>
+        public string Destination { get; set; }
```

Comment:
This one is a little different because the class being used on Unix is different than on Windows, so the implementation is a little different.  I added the aliases in for the above mentioned because on the Unix class the IPAddress class takes these all.